packages/flutter/lib/src/widgets/_web_selectable_region_context_menu_io.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Fake interface for the `WebSelectableRegionContextMenu` widget that requires
+// html library. See `_web_selectable_region_context_menu_web.dart` for web
+// implementation.
+
+import 'framework.dart';
+import 'selection_container.dart';
+
+/// Function signature for `ui.platformViewRegistry.registerViewFactory`.
+@visibleForTesting
+typedef RegisterViewFactory = void Function(String, Object Function(int viewId), {bool isVisible});
+
+/// A mock widget for non-web platform.
+///
+/// See `_web_selectable_region_context_menu_web.dart` for the widget used in web
+/// platform.
+class WebSelectableRegionContextMenu extends StatelessWidget {
+  /// See `_web_selectable_region_context_menu_web.dart`.
+  // ignore: prefer_const_constructors_in_immutables
+  WebSelectableRegionContextMenu({
+    // ignore: avoid_unused_constructor_parameters
+    required Widget child,
+    super.key,
+  });
+
+  /// See `_web_selectable_region_context_menu_web.dart`.
+  static void attach(SelectionContainerDelegate client) => throw UnimplementedError();
+
+  /// See `_web_selectable_region_context_menu_web.dart`.
+  static void detach(SelectionContainerDelegate client) => throw UnimplementedError();
+
+  /// See `_web_selectable_region_context_menu_web.dart`.
+  @visibleForTesting
+  static RegisterViewFactory factory = (String viewType, Object Function(int viewId) fn, {bool isVisible = true}) { };
```

Comment:
Since the test is itself behind the conditional import it will only use the web version. I think you don't need the `io` versions of the `factory` and `typedef RegisterViewFactory`.